### PR TITLE
blockdev_inc_backup_xpt_nonexist_bitmap:Expose non-existed bitmap via qemu-nbd

### PR DESCRIPTION
**Pull request by aliang123**
Expose non-existed bitmap via qemu-nbd

Signed-off-by: Aihua Liang <aliang@redhat.com>
id:1985277

### CODE REVIEW

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_xpt_nonexist_bitmap.q35: PASS (2.84 s)

**Comment by aliang123** [comment]
@zhencliu @ngu-niny, please help to review, thanks.

File: qemu/tests/blockdev_inc_backup_xpt_nonexist_bitmap.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import logging
+import string
+
+from provider.blockdev_base import BlockdevBaseTest
+from provider.nbd_image_export import QemuNBDExportImage
+
+from virttest import utils_misc
+
+from avocado.utils import process
```
Comment (zhencliu) [inline qemu/tests/blockdev_inc_backup_xpt_nonexist_bitmap.py:9]: Better to move these after standard modules:
standard
avocado
vt
tp-qemu

File: qemu/tests/blockdev_inc_backup_xpt_nonexist_bitmap.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+        logging.info("Export inconsistent bitmap with qemu-nbd")
+        cmd = _nbd_expose_cmd(self.nbd_exports[0]._qemu_nbd,
+                              self.nbd_exports[0]._local_filename,
+                              self.nbd_exports[0]._tag,
+                              self.nbd_exports[0]._image_params)
+        result = process.run(cmd, ignore_status=True, shell=True,
+                             ignore_bg_processes=True)
+        error_msg = self.params.get("error_msg") % self.bitmaps[0]
+        if error_msg not in result.stderr.decode():
```
Comment (zhencliu) [inline qemu/tests/blockdev_inc_backup_xpt_nonexist_bitmap.py:66]: I am afraid we may also need to check result.exit_status == 0, i.e. qemu-nbd completed unexpectedly, in this situation, we have to remove the pid file, the most important thing we have to do is kill qemu-nbd, for which is always occupying the port 10850

File: qemu/tests/blockdev_inc_backup_xpt_nonexist_bitmap.py (hunk 1 above)
Comment (aliang123) [reply qemu/tests/blockdev_inc_backup_xpt_nonexist_bitmap.py]: Have updated, thanks zhenchao.

File: qemu/tests/blockdev_inc_backup_xpt_nonexist_bitmap.py (hunk 2 above)
Comment (aliang123) [reply qemu/tests/blockdev_inc_backup_xpt_nonexist_bitmap.py:74]: @zhencliu, have updated code in this section, please help to check.

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_xpt_nonexist_bitmap.q35: PASS (2.85 s)

**Comment by ngu-niny** [comment]
05-Host_RHEL.m8.u5.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.5.0.ppc64le.io-github-autotest-qemu.blockdev_inc_backup_xpt_nonexist_bitmap Finshed 2021-07-29 03:26:04 2021-07-29 03:26:08 PASS 3.861149

07-Host_RHEL.m8.u5.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_inc_backup_xpt_nonexist_bitmap Finshed 2021-07-29 03:28:36 2021-07-29 03:28:44 PASS 8.535502

09-Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.ppc64le.io-github-autotest-qemu.blockdev_inc_backup_xpt_nonexist_bitmap Finshed 2021-07-29 03:31:27 2021-07-29 03:31:35 PASS 8.586657

11-Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_inc_backup_xpt_nonexist_bitmap Finshed 2021-07-29 03:33:43 2021-07-29 03:33:51 PASS 8.25251

Acked-by: Nini Gu <ngu@redhat.com>

**Comment by zhencliu** [comment]
LGTM

**Pull request merged**